LinesCounter/LinesCounter/FilesDirectory.cpp

```diff
@@ -0,0 +1,65 @@
+#include"FilesDirectory.h"
+
+vector<fs::path> ListFilesInDirectory(const fs::path& dir)// get list of files we'll parse
+{
+	vector<fs::path> files;
+	if (fs::exists(dir) && fs::is_directory(dir))// while we have files in the directory
+	{
+		for (const auto& entry : fs::recursive_directory_iterator(dir))// move recoursive throught all directories
+		{
+			if(PropriateExtension(entry))// if our condition is ok (.cpp, .h. etc)
+				files.emplace_back(entry.path());//add this path
+		}
+	}
+	return files;
+}
+
+bool PropriateExtension(fs::path path)//condition
+{
+	vector<fs::path>extension = { ".cpp",".c",".hpp",".h" };
+	for (auto &i : extension)
+	{
+		if (path.extension() == i)
+			return true;
+	}
+	return false;
+}
+
+
+std::string InputPath()
+{
+	std::string path;
+	std::cout << "Input full path: ";
+	std::cin >> path;
+	return path;
+}
+
+void PrintResultInTheFile(std::queue<CodeFile> codeFileQ, std::chrono::microseconds runningDuration)// sum the amount of any type, create file, show it
+{
+	struct LinesCount
+	{
+		int m_processedCount = 0;
+		int m_blankCount = 0;
+		int m_commentCount = 0;
+		int m_physicalCount = 0;
+	}linesCount;
+	int fileParsed = codeFileQ.size();
+
+	while (!codeFileQ.empty())
+	{
+		linesCount.m_processedCount += codeFileQ.front().GetProcessedCount();
+		linesCount.m_physicalCount += codeFileQ.front().GetPhysicalCount();
+		linesCount.m_commentCount += codeFileQ.front().GetCommentCount();
+		linesCount.m_blankCount += codeFileQ.front().GetBlankCount();
+		codeFileQ.pop();
+	}
```

Comment:
I would suggest to overload operator+= for class CodeFile
It would help us to aggregate addition results without calling separate get operations:

CodeFile result;
while(!codeFileQ.empty())
    { results += codeFileQ.front(); codeFileQ.pop(); }